adapter/network_management.md

```diff
@@ -0,0 +1,253 @@
+# Network Management
+
+The following document describes the interaction flows and FIMP specification used in management of adapter network, specifically to get connection status or update the network.
+
+## Behaviors and Flows
+
+### Reporting Network Status
+
+* At any time, upon receiving `cmd.network.get_node` command, an adapter must respond with an appropriate `evt.network.node_report` report containing status of a requested thing.
+* At any time, upon receiving `cmd.network.get_all_nodes` command, an adapter must respond with `evt.network.all_nodes_report` report containing status of all things.
+* On its own, an adapter must publish `evt.network.node_report` whenever the status of a thing changes.
+* On its own, an adapter should publish `evt.network.all_nodes_report` at least once per day.
+
+### Resetting Network
+
+* Upon receiving `cmd.network.reset` command, the adapter should start resetting the network.
+* For each removed thing it should emit `evt.thing.exclusion_report` as specified in [thing management](/adapter/thing_management.md#interfaces) document.
+* Once the process is finished, the adapter should emit `evt.network.reset_done`.
+
+## FIMP Specification
+
+### Topics
+
+Every adapter should have unique resource name and use the following topics in accordance with [topic format](/fimp/topic_format.md):
+
+`pt:j1/mt:cmd/rt:ad/rn:{RESOURCE_NAME}/ad:1`
+
+`pt:j1/mt:evt/rt:ad/rn:{RESOURCE_NAME}/ad:1`
+
+### Service name
+
+Every adapter should define its own service name.
+
+### Interfaces
+
+| Type | Interface                    | Value type | Description                                                                                                       |
+|------|------------------------------|------------|-------------------------------------------------------------------------------------------------------------------|
+| in   | cmd.network.get_node         | string     | Requests state of a thing with the provided address.                                                              |                         
+| out  | evt.network.node_report      | object     | Reports state of a single thing, see [`node_report`](#definitions) object.                                        |
+| in   | cmd.network.get_all_nodes    | null       | Requests state of all things.                                                                                     |                         
+| out  | evt.network.all_nodes_report | object     | Reports state of all things, an array of [`node_report`](#definitions) objects.                                   |
+| in   | cmd.network.update           | string     | Requests update of the network topology, see [`network_update_mode`](#definitions) definition for allowed values. | 
+| in   | cmd.network.node_update      | int        | Requests an interview and discovery of its capabilities for the provided address of a thing.                      |
```

Comment:
Issue 1: `cmd.network.node_update` interface breaks from accepted fact that thing address is elsewhere referenced as a string. While Z-Wave node ID and Zigbee UDID are numeric and in Cliffhanger I have introduced internal mapping between public numeric addresses and internal freeform text identifiers (like UUID used by a cloud system of a third party), there are still some old adapters using non-numeric identifiers for thing address.

We have two options:
* Turn our eyes away from this inconsistency and pretend we did not notice it
* Specify in the documentation that it is string but pledge eternal support in the zigbee and z-wave adapter for both integer and string input on this interface handler. This interface is already implemented this way in Z-Wave.

I am specifically asking @GenixPL  @BudzikM  @eternalamenity @chribru for input on this issue.

Reply:
<img width="1301" alt="image" src="https://user-images.githubusercontent.com/31538120/224342955-b83db2b2-5730-405b-8c7d-c20c61386a17.png">

Reply:
when it comes to the app we dont have much problems with playing around the types (it's a messy everywhere so one more place doesnt matter). we would have substantial amount of work, and complicate things, if we were needed to create two different requests based on some version (plus it can lead to bugs), so i would omit that as much as possible

Reply:
The solution I see is:
* Z-Wave and Zigbee current/old versions accept only integers
* Z-Wave and ZIgbee new versions will accept both strings and integers
* Mobile app will be sending integers to Z-Wave/Zigbee to maintain compatibility and string to other adapters (currently no other adapter supports it).

Considering that It might be unnecessary purism.

Reply:
Lets switch to int and keep consistent

Reply:
I think we should stick with string, because of other adapters or playground/integration apps that do not need to use integers as it's addresses, correct @boguslaw-wojcik ?

Reply:
Yes., @eternalamenity